File: test/single-language/ruby/bundler/http-service/success-runtime-config/lib/configuration.rb
module Fixture
  class Configuration
    attr_reader :mode, :status, :label
    def initialize
      @mode = 'config'
      @status = 200
      @label = @mode == 'config' ? ENV.fetch('FIXTURE_LABEL', 'runtime-config-default') : 'deployment-smoke-ok'
    end
    def port
      raw = ENV.fetch('PORT', '')
      raise ArgumentError, 'Invalid PORT' unless raw.match?(/\A[0-9]+\z/) && (1..65535).cover?(raw.to_i)
      raw.to_i
    end
  end
end

File: test/single-language/node/yarn/typescript-service/success-deployment-smoke/src/configuration.ts
export interface Configuration {
  port: number; mode: string; status: number; label: string;
}
export function loadConfiguration(): Configuration {
  const raw = process.env.PORT ?? '';
  const port = Number(raw);
  if (!/^[0-9]+$/.test(raw) || !Number.isInteger(port) || port < 1 || port > 65535) throw new Error('Invalid PORT');
  const mode: string = 'smoke';
  const label = mode === 'config' ? (process.env.FIXTURE_LABEL ?? 'runtime-config-default') : 'deployment-smoke-ok';
  return { port, mode, status: 200, label };
}
